test/tools/events/Event.java

```diff
@@ -33,8 +48,100 @@ public Event(EventType eventType, IonType ionType, SymbolToken fieldName, Symbol
         this.depth = depth;
     }
 
+    public void validate() throws IonException, IOException {
+        if (this.eventType == null) throw new IonException("event_type can't be null");
+        else if (this.ionType == null && this.eventType != EventType.STREAM_END)
```

Comment:
Added.

I noticed that ion-c's CONTAINER_END event has ion_type but it doesn't affect processing. 
For example I change CONTAINER_END event's ionType from a struct to list, it will still write the same output.